clients/csharp-uwp/UWPVoiceAssistantSample/DialogManager.cs

```diff
@@ -364,7 +364,6 @@ private async Task ChangeAgentStateAsync(ConversationalAgentState newState)
             var oldState = session.AgentState;
             this.logger.Log($"Changing agent state: [{oldState.ToString()}] -> [{newState.ToString()}]");
             await session.RequestAgentStateChangeAsync(newState);
-            this.signalDetectionHelper.DialogStateChangeDuringSignalVerification(oldState, newState);
```

Comment:
Was this already getting handled via another means? It's not following for me how the cancelation path for verification gets removed by the change; I'm sure I'm just missing the logic, though.

Reply:
This was leading to a duplicate signalrejected event. I removed it when trying to debug that bug caused by stop recognition. The only time the state changes from detecting to inactive is when the signaldetectionhelper fires signal rejected, which also triggers a signal rejected call.

The code path was signaldetectionhelper.signalrejected->dialogmanager event handler fires signalrejected (1st time) and changes state from detecting to inactive->signaldetectionhelper's dialogstatechangeduringsignalverification is called with the change from detecting to inactive and fires signalrejected->dialogmanager event handler fires signalrejected (2nd time) and changes state from inactive to inactive->signaldetectionhelper's dialogstatechangeduringsignalverification is called with the change from inactive to inactive and does nothing